Plugin.pm

```diff
@@ -1208,8 +1257,29 @@ sub _trackItem {
 		$item->{line1} .= ' (' . cstring($client, 'PLUGIN_QOBUZ_HIRES') . ')';
 	}
 
-	if ( $track->{work} ) {
-		$item->{work} = $track->{work};
+	# Enhancements to work/composer display for classical music (tags returned from Qobuz are all over the place)
+	my $genreList = $prefs->get('classicalGenres');
+	if ( $prefs->get('useClassicalEnhancements')
+		&& ( grep(/Classique/,@{$track->{album}->{genres_list}}) || grep(/(^\s*$genre\s*$|^\s*\*$)/,(split ',', $genreList )) ) 
+	   ) 
+	{
+		if ( $track->{work} ) {
+#			if ( $track->{work} ne $track->{title} ) {
+				$item->{work} = $track->{work};
+#			}
+		} else {
+			# Try to extract the work from the title
+			if (index($track->{composer}->{name}, (split ':', $track->{title})[0]) == -1) {
```

Comment:
Again: split by regex, check for the existence of "deep" attributes like `$track->{composer}->{name}`.